### ISSUE TITLE
Optional CTS profiles are marking FNS as unsuccessful (and non-conformant)

### ISSUE THREAD

**Issue by cmgrote** [odpi/egeria#2076]
The following optional CTS profiles / test cases still appear to be marking a `FunctionNotSupportedException` as an unexpected exception and therefore an unsuccessful outcome (and the repository therefore non-conformant), rather than as a successful outcome indicating there is not support for that profile (and therefore the repository is conformant, but simply does not support that optional profile):

- [x] `TestSupportedReferenceCopyClassificationLifecycle` (2nd line of code under `run()` method does an `addEntity` that is not try-catch-wrapped); example output:

```json
{
  "profileId": 1,
  "requirementId": 14,
  "testCaseId": "repository-reference-copy-classification-lifecycle-SubjectArea-ConnectorType",
  "testCaseName": "Repository reference copy classification lifecycle test case",
  "testCaseDescriptionURL": "https://egeria.odpi.org/open-metadata-conformance-suite/docs/repository-workbench/test-cases/repository-reference-copy-classification-lifecycle-test-case.md",
  "testEvidenceType": "UNEXPECTED_EXCEPTION",
  "assertionMessage": "test-case-base-01: Unexpected Exception FunctionNotSupportedException",
  "conformanceException": {
    "exceptionClassName": "org.odpi.openmetadata.repositoryservices.ffdc.exception.FunctionNotSupportedException",
    "errorMessage": "OMRS-METADATA-COLLECTION-501-001 OMRSMetadataInstanceStore method addEntity for OMRS Connector org.odpi.egeria.connectors.ibm.igc.repositoryconnector.IGCOMRSMetadataCollection to repository type myserver is not implemented"
  }
}
```

- [x] `TestSupportedEntityLifecycle` (2nd line of code under `run()` method does an `addEntity` that is not try-catch-wrapped); example output:

```json
{
  "profileId": 2,
  "requirementId": 17,
  "testCaseId": "repository-entity-lifecycle-ConnectorType",
  "testCaseName": "Repository entity lifecycle test case",
  "testCaseDescriptionURL": "https://egeria.odpi.org/open-metadata-conformance-suite/docs/repository-workbench/test-cases/repository-entity-lifecycle-test-case.md",
  "testEvidenceType": "UNEXPECTED_EXCEPTION",
  "assertionMessage": "test-case-base-01: Unexpected Exception FunctionNotSupportedException",
  "conformanceException": {
    "exceptionClassName": "org.odpi.openmetadata.repositoryservices.ffdc.exception.FunctionNotSupportedException",
    "errorMessage": "OMRS-METADATA-COLLECTION-501-001 OMRSMetadataInstanceStore method addEntity for OMRS Connector org.odpi.egeria.connectors.ibm.igc.repositoryconnector.IGCOMRSMetadataCollection to repository type myserver is not implemented"
  }
}
```

- [x] `TestSupportedClassificationLifecycle` (2nd line of code under `run()` method calls `addEntityToRepository`, which itself calls `addEntity` without any wrapping in a try-catch); example output:

```json
{
  "profileId": 2,
  "requirementId": 18,
  "testCaseId": "repository-classification-lifecycle-SubjectArea-ConnectorType",
  "testCaseName": "Repository classification lifecycle test case",
  "testCaseDescriptionURL": "https://egeria.odpi.org/open-metadata-conformance-suite/docs/repository-workbench/test-cases/repository-classification-lifecycle-test-case.md",
  "testEvidenceType": "UNEXPECTED_EXCEPTION",
  "assertionMessage": "test-case-base-01: Unexpected Exception FunctionNotSupportedException",
  "conformanceException": {
    "exceptionClassName": "org.odpi.openmetadata.repositoryservices.ffdc.exception.FunctionNotSupportedException",
    "errorMessage": "OMRS-METADATA-COLLECTION-501-001 OMRSMetadataInstanceStore method addEntity for OMRS Connector org.odpi.egeria.connectors.ibm.igc.repositoryconnector.IGCOMRSMetadataCollection to repository type myserver is not implemented"
  }
}
```

- [x] `TestSupportedRelationshipLifecycle` (5th line of code under `run()` method calls `addRelationship` that is not try-catch-wrapped); example output:

```json
{
  "profileId": 2,
  "requirementId": 19,
  "testCaseId": "repository-relationship-lifecycle-AttachedTag",
  "testCaseName": "Repository relationship lifecycle test case",
  "testCaseDescriptionURL": "https://egeria.odpi.org/open-metadata-conformance-suite/docs/repository-workbench/test-cases/repository-relationship-lifecycle-test-case.md",
  "testEvidenceType": "UNEXPECTED_EXCEPTION",
  "assertionMessage": "test-case-base-01: Unexpected Exception FunctionNotSupportedException",
  "conformanceException": {
    "exceptionClassName": "org.odpi.openmetadata.repositoryservices.ffdc.exception.FunctionNotSupportedException",
    "errorMessage": "OMRS-METADATA-COLLECTION-501-001 OMRSMetadataInstanceStore method addEntity for OMRS Connector org.odpi.egeria.connectors.ibm.igc.repositoryconnector.IGCOMRSMetadataCollection to repository type myserver is not implemented"
  }
}
```

- [x] `TestSupportedEntityReidentify` (2nd line of code under `run()` method calls `addEntity` that is not try-catch-wrapped); example output:

```json
{
  "profileId": 9,
  "requirementId": 34,
  "testCaseId": "repository-entity-reidentify-ConnectorType",
  "testCaseName": "Repository entity reidentify test case",
  "testCaseDescriptionURL": "https://egeria.odpi.org/open-metadata-conformance-suite/docs/repository-workbench/test-cases/repository-entity-reidentify-test-case.md",
  "testEvidenceType": "UNEXPECTED_EXCEPTION",
  "assertionMessage": "test-case-base-01: Unexpected Exception FunctionNotSupportedException",
  "conformanceException": {
    "exceptionClassName": "org.odpi.openmetadata.repositoryservices.ffdc.exception.FunctionNotSupportedException",
    "errorMessage": "OMRS-METADATA-COLLECTION-501-001 OMRSMetadataInstanceStore method addEntity for OMRS Connector org.odpi.egeria.connectors.ibm.igc.repositoryconnector.IGCOMRSMetadataCollection to repository type myserver is not implemented"
  }
}
```

- [x] `TestSupportedEntityRetype` (3rd line of code under `run()` method calls `addEntity` that is not try-catch-wrapped); example output:

```json
{
  "profileId": 10,
  "requirementId": 37,
  "testCaseId": "repository-entity-retype-ConnectorType",
  "testCaseName": "Repository entity retype test case",
  "testCaseDescriptionURL": "https://egeria.odpi.org/open-metadata-conformance-suite/docs/repository-workbench/test-cases/repository-entity-retype-test-case.md",
  "testEvidenceType": "UNEXPECTED_EXCEPTION",
  "assertionMessage": "test-case-base-01: Unexpected Exception FunctionNotSupportedException",
  "conformanceException": {
    "exceptionClassName": "org.odpi.openmetadata.repositoryservices.ffdc.exception.FunctionNotSupportedException",
    "errorMessage": "OMRS-METADATA-COLLECTION-501-001 OMRSMetadataInstanceStore method addEntity for OMRS Connector org.odpi.egeria.connectors.ibm.igc.repositoryconnector.IGCOMRSMetadataCollection to repository type myserver is not implemented"
  }
}
```

**Comment by grahamwallis** [comment]
Thanks - I'm on the case

**Comment by cmgrote** [comment]
Resolved by #2102 and #2093
